pylint/checkers/typecheck.py

```diff
@@ -1058,6 +1064,39 @@ def _check_uninferable_call(self, node):
                     )
                     break
 
+    def _check_argument_order(self, node, call_site, called, called_param_names):
+        """Match the supplied argument names against the function parameters.
+        Warn if some argument names are not in the same order as they are in
+        the function signature.
+        """
+        # Check for called function being an object instance function
+        # If so, ignore the initial 'self' argument in the signature
+        try:
+            if (isinstance(called.parent, astroid.scoped_nodes.ClassDef)
+                and called_param_names[0] == 'self'):
```

Comment:
Is there a better way of checking for the presense of a `self`-type attribute in the args list? This one will fail if the user is calling `self` by a different name, although a rare pattern to follow anyway.